asv/environment.py

```diff
@@ -428,18 +438,24 @@ def get_environment_class(conf, python):
         if cls.matches_python_fallback and cls.matches(python):
             return cls
     raise EnvironmentUnavailable(
-        "No way to create environment for python='{0}'".format(python))
+        "No way to create environment for python='{0}'".format(python)
+    )
 
 
 def get_environment_class_by_name(environment_type):
     """
     Find the environment class with the given name.
     """
     for cls in util.iter_subclasses(Environment):
+        if "mamba" in environment_type and (
+            sys.version_info.major >= 3 and sys.version_info.minor < 8
+        ):
+            environment_type = "conda"
```

Comment:
why the upper limit? Also: what happens if "conda" is used instead of "mamba"?

Reply:
If `conda` is used it continues to use the old solver. The idea here was that it should silently fall back to the `conda` mechanism before `libmambapy` is present. The upper limit shouldn't be there, sorry.